sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -105,77 +244,212 @@ def hpat_pandas_series_getitem(self, idx):
     _func_name = 'Operator getitem().'
 
     if not isinstance(self, SeriesType):
-        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+        return None
 
-    if isinstance(idx, types.Integer):
-        def hpat_pandas_series_getitem_idx_integer_impl(self, idx):
-            """
-            **Test**: python -m sdc.runtests sdc.tests.test_series.TestSeries.test_series_iloc1
-            """
+    # Note: Getitem return Series
+    index_is_none = self.index is None or isinstance(self.index, numba.types.misc.NoneType)
+    index_is_number = index_is_none or (self.index and isinstance(self.index.dtype, types.Number))
+    index_is_string = not index_is_none and isinstance(self.index.dtype, (types.UnicodeType, types.StringLiteral))
+
+    if (
+        isinstance(idx, types.Number) and index_is_number or
+        (isinstance(idx, (str, types.UnicodeType, types.StringLiteral)) and index_is_string)
+    ):
+        def hpat_pandas_series_getitem_index_impl(self, idx):
+            mask = numpy.empty(len(self._data), numpy.bool_)
+            for i in numba.prange(len(self.index)):
+                mask[i] = self.index[i] == idx
+            return pandas.Series(self._data[mask], self.index[mask])
 
-            result = self._data[idx]
-            return result
+        return hpat_pandas_series_getitem_index_impl
 
-        return hpat_pandas_series_getitem_idx_integer_impl
+    if (isinstance(idx, types.Integer) and index_is_string):
+        def hpat_pandas_series_idx_impl(self, idx):
+            return self._data[idx]
+
+        return hpat_pandas_series_idx_impl
 
     if isinstance(idx, types.SliceType):
+        # Return slice for str values not implement
```

Comment:
Actually it's implemented (might not be working though), see (in str_arr_ext.py):
```
@lower_builtin(operator.getitem, StringArrayType, types.SliceType)
def lower_string_arr_getitem_slice(context, builder, sig, args):
```